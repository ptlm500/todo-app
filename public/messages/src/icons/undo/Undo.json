[
  {
    "id": "todo.icons.undo.title",
    "description": "Undo icon title",
    "defaultMessage": "Undo"
  }
]