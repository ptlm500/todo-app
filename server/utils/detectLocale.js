const acceptLanguage = require('accept-language');
const config = require('../config.json');

function detectLocale(req) {
  acceptLanguage.languages(config.ACCEPTED_LANGUAGES);

  const cookieLocale = req.cookies.locale;

  return acceptLanguage.get(cookieLocale || req.headers['accept-language']) ||
    config.DEFAULT_LOCALE;
}

module.exports = detectLocale;
