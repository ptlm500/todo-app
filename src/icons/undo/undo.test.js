import React from 'react';
import { renderWithIntl } from '../../../test/utils';
import Undo from './Undo';

describe('Undo', () => {
  it('renders a Undo icon', () => {
    const tree = renderWithIntl(<Undo />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a Undo icon with custom title', () => {
    const tree = renderWithIntl(<Undo title={'test title'}/>);

    expect(tree.root.findByType('title').children).toEqual(['test title']);
    expect(tree).toMatchSnapshot();
  });
});
