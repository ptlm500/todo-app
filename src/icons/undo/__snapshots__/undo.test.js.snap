// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Undo renders a Undo icon 1`] = `
<svg
  height="9px"
  version="1.1"
  viewBox="0 0 21 9"
  width="21px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Undo
  </title>
  <g
    fill="none"
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      transform="translate(-238.000000, -1533.000000)"
    >
      <g
        transform="translate(100.000000, 1428.000000)"
      >
        <g
          id="Outlined-/-Content-/-undo"
          transform="translate(136.000000, 98.000000)"
        >
          <g>
            <polygon
              id="Path"
              points="0 0 24 0 24 24 0 24"
            />
            <path
              d="M12.5,8 C9.85,8 7.45,8.99 5.6,10.6 L2,7 L2,16 L11,16 L7.38,12.38 C8.77,11.22 10.54,10.5 12.5,10.5 C16.04,10.5 19.05,12.81 20.1,16 L22.47,15.22 C21.08,11.03 17.15,8 12.5,8 Z"
              fill="currentColor"
            />
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`Undo renders a Undo icon with custom title 1`] = `
<svg
  height="9px"
  version="1.1"
  viewBox="0 0 21 9"
  width="21px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    test title
  </title>
  <g
    fill="none"
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      transform="translate(-238.000000, -1533.000000)"
    >
      <g
        transform="translate(100.000000, 1428.000000)"
      >
        <g
          id="Outlined-/-Content-/-undo"
          transform="translate(136.000000, 98.000000)"
        >
          <g>
            <polygon
              id="Path"
              points="0 0 24 0 24 24 0 24"
            />
            <path
              d="M12.5,8 C9.85,8 7.45,8.99 5.6,10.6 L2,7 L2,16 L11,16 L7.38,12.38 C8.77,11.22 10.54,10.5 12.5,10.5 C16.04,10.5 19.05,12.81 20.1,16 L22.47,15.22 C21.08,11.03 17.15,8 12.5,8 Z"
              fill="currentColor"
            />
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;
